pkg/ottl/ottlfuncs/func_delete_key.go

```diff
@@ -22,9 +22,26 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
-func DeleteKey[K any](target ottl.Getter[K], key string) (ottl.ExprFunc[K], error) {
+type DeleteKeyArguments[K any] struct {
+	Target ottl.Getter[K] `ottlarg:"0"`
+	Key    string         `ottlarg:"1"`
+}
+
+type DeleteKeyFactory[K any] struct{}
+
+func (f DeleteKeyFactory[K]) Name() string { return "delete_key" }
+
+func (f DeleteKeyFactory[K]) CreateEmptyArguments() ottl.Arguments {
+	return &DeleteKeyArguments[K]{}
+}
+
+func (f DeleteKeyFactory[K]) CreateFunction(ctx context.Context, fCtx ottl.FunctionContext, args ottl.Arguments) (ottl.ExprFunc[K], error) {
+	return deleteKey(ctx, fCtx, args.(*DeleteKeyArguments[K])), nil
+}
+
+func deleteKey[K any](_ context.Context, fCtx ottl.FunctionContext, args *DeleteKeyArguments[K]) ottl.ExprFunc[K] {
```

Comment:
I think I like the way this looks.  I helps differentiate between the function itself and the Factory that is providing it to OTTL.  Might just be more boilerplate tho.

Reply:
I'm fine with it since it's a little more concise to read the actual functionality, but it's not all that much different than having the definition in `CreateFunction`, so I could go either way. I'll write some tests and see if it alleviates anything there.

Reply:
I'd say that if it doesn't make testing any easier then we don't need it. it is kinda silly to pass all the args to a second function unless we need tangibly need it.  I don't think having a function literally named `deleteKey` makes understanding the contents of `delete_key.go` that much easier to understand, probably not worth the boilerplate.

Reply:
I played around with it, and I don't think it will provide enough benefit to justify the extra code. I've put the body back in `CreateFunction`.